accepted/2021/experimental-features/experimental-features.md

```diff
@@ -0,0 +1,380 @@
+computations.
+
+As part of their research into this problem space, they notice a recent blog
+post about how the upcoming generic numeric support and hardware intrinsics can
+help to speed things up considerably. They decide to give it a spin, so they
+copy & paste some sample code from the blog post.
+
+The code immediately produces an error message:
+
+> error: The type 'IArithmetic\<T>' is part of the experimental feature
+> 'StaticInterfaceMembers' which must be turned on explicitly.
+
+After taking a quick look at the blog post, Ainsley realizes that they only need
+to set the `ExperimentalFeatures` property in the project files.
+
+### Consuming a library that uses an experimental feature
+
+Sahibi works with Ainsley and tries to consume their library that exposes some
+of the core algorithms so she can integrate those with ML.NET. After adding the
+reference to Ainsley's library `Fabrikam.Algorithms.Core` she instantiates the
+engine type which produces the following error:
+
+> The library 'Fabrikam.Algorithms.Core' requires the experimental feature
+> 'StaticInterfaceMembers' which must be turned on explicitly.
+
+Sahibi IMs Ainsley to ask what this is about where she learns that this feature
+isn't a supported part of .NET, therefore she should make sure to not integrate
+the library into the production instance of Fabrikam.
+
+## Requirements
+
+### Goals
+
+* Customers can discover experimental features
+* Customers are made aware if a particular feature is in an experimental state
+  so that they always know which part of the product is fully supported and
+  which parts aren't.
+* Customer have an easy way to audit if their applications or libraries are
+  using experimental features.
+* Customers can't accidentally depend on experimental features, either directly
+  or indirectly through some library.
+* We can deliver a reasonable experience even when an experimental feature can't
+  be supported in all the workloads, for example, by failing with a sensible
+  error message.
+* We have a unified experience for turning on experimental features, no matter
+  whether they are in a runtime, a compiler, a library, or span multiple parts
+  of the system.
+* We don't have to do unnatural acts to add an experimental feature, by, for
+  example, having to build a plug-in based system that allows swapping in and
+  out files.
+
+### Non-Goals
+
+* Allowing Microsoft to turn features on or off remotely, e.g. A/B testing
+* Designing a mechanism for shipping experimental features as separate packages
+  or installers
+
+## Stakeholders and Reviewers
+
+* Runtime teams
+* C#, VB, F# teams
+* Library teams
+* SDK team
+* MSBuild team
+* NuGet team
+
+## Design
+
+### Property in project files
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <ExperimentalFeatures>StaticInterfaceMembers</ExperimentalFeatures>
```

Comment:
Shouldn't this be expressed using MSBuild items instead of properties?  For example:

```xml
<ItemGroup>
    <EnableExperimentalFeature Include="StaticInterfaceMembers" />
</ItemGroup>
```

Expressing multiple values as a semicolon-separated list for a property isn't the greatest.

Reply:
Fixed